t/22-svirt.t

```diff
@@ -499,6 +499,14 @@ sub svirt_xml_validate ($svirt, %args) {
         is($driver_node->getAttribute('type'), $args{driver}->{type}, 'type attribute of <driver> is correct');
         is($driver_node->getAttribute('cache'), $args{driver}->{cache}, 'cache attribute of <driver> is correct');
     }
+
+    if($args{usb}) {
+        #<controller type="usb" index="0" ports="8"/>
+        my $target = $doc->findnodes('domain/devices/controller')->shift();
+        my $driver_node = $target->parentNode->findnodes('controller')->shift;
+        is($driver_node->getAttribute('ports'), 8, 'no. of ports <controller> is correct');
+        is($driver_node->getAttribute('type'), 'usb', 'type attribute of <controller> is correct');
+    }
```

Comment:
```suggestion
    subtest '<controller type="usb" index="0" ports="8"/>' => sub {
        my $target = $doc->findnodes('domain/devices/controller')->shift();
        my $driver_node = $target->parentNode->findnodes('controller')->shift;
        is($driver_node->getAttribute('ports'), 8, 'no. of ports <controller> is correct');
        is($driver_node->getAttribute('type'), 'usb', 'type attribute of <controller> is correct');
    } if $args{usb};
```

Reply:
Thanks Marius, you have inspired me to move this code as a subtest for xen-pv creation.